Source/Core/InputCommon/ControllerInterface/Device.cpp

```diff
@@ -26,45 +27,38 @@ namespace Core
 //
 Device::~Device()
 {
-  // delete inputs
-  for (Device::Input* input : m_inputs)
-    delete input;
-
-  // delete outputs
-  for (Device::Output* output : m_outputs)
-    delete output;
 }
 
-void Device::AddInput(Device::Input* const i)
+void Device::AddInput(std::unique_ptr<Input> input)
 {
-  m_inputs.push_back(i);
+  m_inputs.push_back(std::move(input));
```

Comment:
Should be good as-is...or i misunderstood this [StackOverflow post](http://stackoverflow.com/a/26861516)?